parseSchema/index.js

```diff
@@ -0,0 +1,111 @@
+"use strict";
+
+const Parse = require('parse/node');
+
+function updateSchemaOnlyWhenNecessary(existingSchemaData, intendedSchema)
+{
+  // remove any existing fields from the request.
+  Object.getOwnPropertyNames(existingSchemaData.fields).forEach(existingField => { delete intendedSchema._fields[existingField] ; });
+  return (Object.getOwnPropertyNames(intendedSchema._fields).length + Object.getOwnPropertyNames(intendedSchema._indexes).length > 0 ) ? 
```

Comment:
it doesn't look like this code removes indices that already exist from the request

Reply:
True... Object.getOwnPropertyNames(intendedSchema._indexes).length is there to make that statement (and that statement only) "future-proof"...